### PR TITLE
ADD: add external caliper delivery API call

### PR DESCRIPTION
**Pull request by asamolion**
UCSD has requested to use an API endpoint as delivery for
caliper endpoint. I have added a very basic API call that handles this
scenario

### CODE REVIEW

File: openedx_caliper_tracking/processor.py [hunk 1]
```diff
@@ -10,16 +12,25 @@
...
         try:
             caliper_event = base_transformer(event)
             related_function = EVENT_MAPPING[event.get('event_type')]
-            caliper_logger.info(json.dumps(related_function(event, caliper_event)))
+            transformed_event = related_function(event, caliper_event)
+            caliper_logger.info(json.dumps(transformed_event))
+            if hasattr(settings, 'CALIPER_DELIVERY_ENDPOINT') and hasattr(settings, 'CALIPER_DELIVERY_AUTH_TOKEN'):
+                response = requests.post(settings.CALIPER_DELIVERY_ENDPOINT,
+                              headers={
+                                  'Authorization': 'Beater {}'.format(settings.CALIPER_DELIVERY_AUTH_TOKEN),
```
Comment (mumarkhan999) [inline openedx_caliper_tracking/processor.py]: I think it is `Bearer` not `Beater`.

**Review by imhassantariq — COMMENTED** [review]
Why we are not handling any response status? What if our event doesn't properly post to API? It will be a loss of data.

**Review by imhassantariq — APPROVED** [review]

File: openedx_caliper_tracking/processor.py (hunk 1 above)
Comment (asamolion) [reply openedx_caliper_tracking/processor.py]: Thanks.

**Comment by asamolion** [comment]
> Why we are not handling any response status? What if our event doesn't properly post to API? It will be a loss of data.

I've added support for response logging.

**Review by tehreem-sadat — APPROVED** [review]

File: openedx_caliper_tracking/processor.py [hunk 2]
```diff
@@ -1,31 +1,123 @@
...
+    This transformer is used in the commong.djangoapps.track django app as a replacement
+    for the default tracking backend.
+
+    It is also used as an addition to the event tracking pipeline in the event_tracking app
+    by Open edX
+    """
+
     def __call__(self, event):
+        """
+        handles the transformatio and delivery of an event. Delivers the event to the caliper log file as well as
```
Comment (HamzaIbnFarooq) [inline openedx_caliper_tracking/processor.py]: transformation

**Review by HamzaIbnFarooq — APPROVED** [review]
:+1:

**Pull request merged**